smartsim/_core/_cli/utils.py

```diff
@@ -81,3 +85,100 @@ def pip_install(packages: t.List[str], end_point: t.Optional[str] = None, verbos
         raise BuildError(error)
     if verbose:
         logger.info(f"{packages} installed successfully")
+
+
+def clean(core_path: Path, _all: bool = False) -> int:
+    """Remove pre existing installations of ML runtimes
+
+    :param _all: Remove all non-python dependencies
+    :type _all: bool, optional
+    """
+
+    build_temp = core_path / ".third-party"
+    if build_temp.is_dir():
+        shutil.rmtree(build_temp, ignore_errors=True)
+
+    lib_path = core_path / "lib"
+    if lib_path.is_dir():
+
+        # remove RedisAI
+        rai_path = lib_path / "redisai.so"
+        if rai_path.is_file():
+            rai_path.unlink()
+            logger.info("Successfully removed existing RedisAI installation")
+
+        backend_path = lib_path / "backends"
+        if backend_path.is_dir():
+            shutil.rmtree(backend_path, ignore_errors=True)
+            logger.info("Successfully removed ML runtimes")
+
+    bin_path = core_path / "bin"
+    if bin_path.is_dir() and _all:
+        files_to_remove = ["redis-server", "redis-cli", "keydb-server", "keydb-cli"]
+        removed = False
+        for _file in files_to_remove:
+            file_path = bin_path.joinpath(_file)
+
+            if file_path.is_file():
+                removed = True
+                file_path.unlink()
+        if removed:
+            logger.info("Successfully removed SmartSim database installation")
+
+    return 0
+
+
+def get_db_path() -> t.Optional[Path]:
+    bin_path = get_install_path() / "_core" / "bin"
+    for option in bin_path.iterdir():
+        if option.name in ("redis-cli", "keydb-cli"):
+            return option
+    return None
+
+
+CliHandler = t.Callable[[Namespace], int]
+CliParseConfigurator = t.Callable[[ArgumentParser], None]
+
+class MenuItemConfig:
+    def __init__(self,
+                 cmd: str,
+                 help: str,
+                 handler: CliHandler,
+                 configurator: t.Optional[CliParseConfigurator] = None):
+        self.command = cmd
+        self.help = help
+        self.handler = handler
+        self.configurator = configurator
+
+    @property
+    def command(self) -> str:
+        return self._cmd
+
+    @command.setter
+    def command(self, value: str) -> None:
+        self._cmd = value
+
+    @property
+    def help(self) -> str:
+        return self._help
+
+    @help.setter
+    def help(self, value: str) -> None:
+        self._help = value
+
+    @property
+    def handler(self) -> CliHandler:
+        return self._handler
+
+    @handler.setter
+    def handler(self, value: CliHandler) -> None:
+        self._handler = value    
+
+    @property
+    def configurator(self) -> t.Optional[CliParseConfigurator]:
+        return self._config
+
+    @configurator.setter
+    def configurator(self, value: t.Optional[CliParseConfigurator]) -> None:
+        self._config = value
```

Comment:
Are these properties strictly needed if we are going to add both read and write access to the underlying data anyway? Would you be against just doing something like
```py
class MenuItemConfig:
    def __init__(...):
        # no "protected" attrs
        self.command = ...
        self.help = ...
        self. handler = ...
        self.configurator = ...

    # No properties on the class
```
The only reason I mention is because this looks like a pretty simple wrapper class around some data  and might be a good candidate to turn into a `dataclass` or something.

Reply:
I think I went through a couple iterations where I was computing/reusing some properties and ended up simplifying it so these have no value. Great eye.